crates/rolldown/tests/rolldown/topics/jsx/preserve/artifacts.snap

```diff
@@ -0,0 +1,24 @@
+---
+source: crates/rolldown_testing/src/integration_test.rs
+---
+# Assets
+
+## main.js
+
+```js
+import { Router as Router$1 } from "@solidjs/router";
+
+
+//#region util.jsx
+var util_exports = {};
+__export(util_exports, { Router: () => Router$2 });
+const Router$2 = () => <div />;
+
+//#endregion
+//#region main.jsx
+const Router = () => <util_exports.Router />;
```

Comment:
I remember esbuild will forces the name to ba capitalized if it's used in JSX. It's invalid to not have capitalized name in JSX.

Reply:
https://hyrious.me/esbuild-repl/?version=0.23.0&b=e%00entry.jsx%00import+*+as+A+from+%22.%2Futil.jsx%22%3B%0A%0Aconsole.log%28%3CA.B.Router+%2F%3E%29%3B%0A%0Aconsole.log%28A.Router%29%0A&b=%00util.jsx%00export+*+as+B+from+%27.%2Ftest.jsx%27&b=%00test.jsx%00export+const+Router+%3D+%28%29+%3D%3E+%3Cdiv+%2F%3E%3B%0A&o=%7B%0A++treeShaking%3A+true%2C%0A%0A%22bundle%22%3A+true%2C%0Aformat%3A+%22esm%22%2C%0Ajsx%3A+%22preserve%22%0A%7D